### PR TITLE
standards control feature added

### PR DESCRIPTION
**Pull request by johnricords**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -8,6 +8,16 @@ variable "action_targets" {
   default = []
 }
 
+variable "standards_control" {
+  description = "List of Security Hub standards to enable or disable in current region."
+  type = list(object({
+    standards_control_arn = string
+    control_status        = string
+    disabled_reason       = string
+  }))
```
Comment (lorengordon) [inline variables.tf:17]: One suggestion. Using for_each over an arn frequently has some odd edge cases, depending on how the module is used. Much prefer to inject a `name` attribute, that the user fully owns and sets up front, even if `name` is required nowhere else except as the for_each label.

```suggestion
  type = list(object({
    name                  = string
    standards_control_arn = string
    control_status        = string
    disabled_reason       = string
  }))
```

The for_each would then be:

```
  for_each = { for control in var.standards_control : control.name => control }
```

Comment (johnricords) [reply variables.tf:17]: name isn't a supported attribute for the resource, so trying to inject that just gives me an unsupported argument

File: tests/securityhub_same_account/main.tf [hunk 2]
```diff
@@ -15,6 +15,27 @@ module "securityhub" {
   ]
 }
 
+module "standards_control" {
+  source = "../../"
+
+  standards_control = [
+    {
+      standards_control_arn = "arn:aws:securityhub:::ruleset/cis-aws-foundations-benchmark/v/1.2.0"
```
Comment (lorengordon) [inline tests/securityhub_same_account/main.tf:23]: Let's add the name here, and also add a name for the second item in the list:

```suggestion
      name                  = "cis-aws-foundations-benchmark"
      standards_control_arn = "arn:aws:securityhub:::ruleset/cis-aws-foundations-benchmark/v/1.2.0"
```

File: main.tf [hunk 3]
```diff
@@ -17,6 +17,16 @@ module "subscriptions" {
   ]
 }
 
+# Manage Control status
+module "standards_control" {
+  source   = "./modules/standards_control"
+  for_each = { for control in var.standards_control : control.name => control }
+
+  standards_control_arn = each.key
```
Comment (lorengordon) [inline main.tf:25]: oh, since we changed the key, this needs to use the arn value:

```suggestion
  standards_control_arn = each.value.arn
```

**Comment by lorengordon** [comment]
Result of `make test`:

```
AWS_DEFAULT_REGION=us-east-1 TF_VAR_member_email=...... make test PYTEST_ARGS="-v --nomock"
[bootstrap]: Using external tardigrade-ci, skipping auto-init
[pytest//home/loren/git/repos/tardigrade-ci/tests/terraform_pytest] Starting Python tests found under the directory "/home/loren/git/repos/tardigrade-ci/tests/terraform_pytest"
pytest /home/loren/git/repos/tardigrade-ci/tests/terraform_pytest -v --nomock
============================================= test session starts ==============================================
platform linux -- Python 3.8.10, pytest-7.2.2, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
rootdir: /home/loren/git/repos
plugins: mock-3.6.1, cov-2.12.1, travis-fold-1.3.0
collected 2 items

../tardigrade-ci/tests/terraform_pytest/test_terraform_install.py::test_modules[securityhub_cross_account] PASSED                                                                                                                                                            [ 50%]
../tardigrade-ci/tests/terraform_pytest/test_terraform_install.py::test_modules[securityhub_same_account] PASSED                                                                                                                                                             [100%]

========================================= 2 passed in 81.63s (0:01:21) =========================================
[pytest//home/loren/git/repos/tardigrade-ci/tests/terraform_pytest]: Tests executed!
```

**Review by lorengordon — APPROVED** [review]

**Pull request merged**